src/xcode/ENA/ENA/Source/Services/HealthCertificate/Models/Extensions/HealthCertificateArray+MostRelevant.swift

```diff
@@ -42,18 +42,34 @@ extension Array where Element == HealthCertificate {
 	}
 
 	var lastCompleteVaccinationCertificate: HealthCertificate? {
-		last {
+		// 1- we filter all the certificates to get an array with only complete Vaccination Certificates
+		let completeVaccinationCertificates = filter {
 			guard let vaccinationEntry = $0.vaccinationEntry else {
 				return false
 			}
-
+			
 			return vaccinationEntry.doseNumber >= vaccinationEntry.totalSeriesOfDoses && (
 				$0.ageInDays ?? 0 > 14 ||
 				vaccinationEntry.isBoosterVaccination ||
-				vaccinationEntry.isRecoveredVaccination)
+				vaccinationEntry.isRecoveredVaccination ||
+				vaccinationEntry.doseNumber > vaccinationEntry.totalSeriesOfDoses
```

Comment:
I think this logic should be included in the `isBoosterVaccination` property, because other places have to be able to rely on it detecting booster vaccinations as well.

And it would be great if you can add the unit tests to the target again for this PR already and update them, they should definitely be included in the target. It should be enough to extend the `vaccinationCertificate(type: .booster, ...)` helper to cover boosters with dn > sd.